presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationImplementation.java

```diff
@@ -268,19 +271,29 @@ public static AggregationImplementation parseImplementation(Class<?> aggregation
                     inputTypes,
                     false);
 
-            return new AggregationImplementation(signature,
-                    aggregationDefinition,
-                    stateClass,
-                    inputFunction,
-                    outputFunction,
-                    combineFunction,
-                    stateSerializerFactoryFunction,
-                    signatureArgumentsTypes,
-                    inputDependencies,
-                    combineDependencies,
-                    outputDependencies,
-                    stateSerializerFactoryDependencies,
-                    parameterTypes);
+            try {
+                Optional<MethodHandle> stateSerializerFactoryFunctionHandle = Optional.empty();
+                if (stateSerializerFactoryFunction.isPresent()) {
+                    stateSerializerFactoryFunctionHandle = Optional.of(lookup().unreflect(stateSerializerFactoryFunction.get()));
+                }
+
+                return new AggregationImplementation(signature,
+                        aggregationDefinition,
+                        stateClass,
+                        lookup().unreflect(inputFunction),
+                        lookup().unreflect(outputFunction),
+                        lookup().unreflect(combineFunction),
```

Comment:
Use `Reflection.methodHandle`

Please search for `unreflect` in all your commits and remove them.